pkg/binding/bindingpolicy.go

```diff
@@ -86,12 +86,17 @@ func (c *Controller) handleBindingPolicy(ctx context.Context, objIdentifier util
 		if err != nil {
 			return fmt.Errorf("failed to ocm.FindClustersBySelectors: %w", err)
 		}
+		if len(clusterSet) == 0 {
+			logger.Info("No clusters are selected by BindingPolicy", "name", bindingPolicy.Name)
```

Comment:
This is a user error _if_ `bindingPolicy.Spec.WantSingletonReportedState`, otherwise it is not a problem. The user always gets feedback on `len(clusterSet)` by counting the listed clusters in the Binding object, so logging it in non-problematic cases seems spurious to me. This can be improved in follow-on work.

/approve

Reply:
IMO, `bindingPolicy.Spec.WantSingletonReportedState` doesn't affect whether `len(clusterSet)==0` is a problem or not.

If it's 'impossible to know whether or not the error is transient' for `WantSingletonReportedState`==true, then it's also 'impossible to know whether or not the error is transient' for `WantSingletonReportedState`==false.

Reply:
`len(clusterSet)==0 && bindingPolicy.Spec.WantSingletonReportedState` is an inconsistent state. The problem is the conjunction. The problem may go away in the future. Now that #2044 has merged we can make progress on state-based error reporting.